### DISCUSSION TITLE
Improve support of the '-p' option

### DISCUSSION THREAD

**Discussion by jinxka** [postmodern/ruby-nmap#96] [Q&A] [answered]
Hello,

The nmap option '-p' allow us to specify whether we want the port to use TCP or UDP like:

`nmap -Pn -sU -sT -p U:102,T:502`

Would it be possible to upgrade the gem to follow this comportement?

Thanks

**Comment by postmodern** [comment, marked as answer]
The `ports` attribute can already accept a Hash of port numbers and Ranges grouped by protocol:

```ruby
nmap.ports = {tcp: [1,2,3,100..200], udp: [1,2,3,100..200]}
```

You can also specify the raw String:

```ruby
nmap.ports = "U:102,T:502"
```

**Reply by jinxka** [reply]
Thanks a lot for the answer !
